pkg/agent/baker_test.go

```diff
@@ -2043,7 +2049,12 @@ func verifyCertsEncoding(cert string) error {
 func getDecodedFilesFromCustomdata(data []byte) (map[string]*decodedValue, error) {
 	var customData cloudInit
 
-	if err := yaml.Unmarshal(data, &customData); err != nil {
+	decodedCse, err := getGzipDecodedValue(data)
+	if err != nil {
+		decodedCse = data
```

Comment:
same here, it's a test. I don't think we should be flexible in what we allow in tests